File: src/main/java/live/einfachgustaf/mods/smp/mixins/MixinPlayerChatMessage.java
package live.einfachgustaf.mods.smp.mixins;

import net.minecraft.network.chat.Component;
import net.minecraft.network.chat.PlayerChatMessage;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;

@Mixin(PlayerChatMessage.class)
public class MixinPlayerChatMessage {

    @Inject(at = @At("RETURN"), method = "decoratedContent", cancellable = true)
    private void injectDecoratedContent(CallbackInfoReturnable<Component> cir) {
        live.einfachgustaf.mods.smp.mixinkt.MixinPlayerChatMessage.INSTANCE.injectDecoratedContent(cir);
    }
}

File: src/main/kotlin/live/einfachgustaf/mods/smp/mixinkt/MixinPlayerChatMessage.kt
package live.einfachgustaf.mods.smp.mixinkt

import net.minecraft.network.chat.Component
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable

object MixinPlayerChatMessage {

    fun injectDecoratedContent(cir: CallbackInfoReturnable<Component>) {
        // TODO: MiniMessage Translation
    }
}